src/software/embedded/gpio_char_dev.cpp

```diff
@@ -0,0 +1,111 @@
+#include "software/embedded/gpio_char_dev.h"
+
+#include <linux/gpio.h>
+
+#include "software/logger/logger.h"
+
+GpioCharDev::GpioCharDev(int gpio_number, GpioDirection direction,
```

Comment:
Maybe change this to an unsigned int, ideally uint32_t since this is system independent.

Reply:
good catch, updated!